syntax = "proto3";

import 'proto/profile.proto';

// 记录相关数据
// 单条记录内容
message RoughRecord {
  string id = 1;
  string user = 2;
  int64 time = 3;
  string weather = 4;
  string mood = 5;
  string status = 6;
  string text = 7;
  string address = 8;
  bool is_public = 9;
  bool is_delete = 10;
  int32 picture_count = 11;
  repeated string pictures = 12;
}

// 记录概要内容
message StripRecord {
  string id = 1;                       // 记录id
  string picture = 2;                  // 记录图片
  string text = 3;                     // 记录文本内容
}

// 记录互动统计
message InterStat {
  int32 view = 1;
  int32 like = 2;
  int32 comment = 3;
}


message ExtRecord {
 RoughRecord record = 1;                // 记录
 StripUserProfile user = 2;             // 用户
 InterStat interact = 3;                // 互动数据，不cache，需单独拉取
}

service RecordServer {
  // Sends a greeting
  rpc HandleRecord (RecordReq) returns (RecordRes) {}
}

enum RecordOP {
  RECORD_OP_INVALID = 0;               // 非法操作
  RECORD_OP_ADD = 1;                   // 新增
  RECORD_OP_DELETE = 2;                // 删除
}

message RecordReq {
  RecordOP op = 1;
  string id = 2;
  int64 time = 3;
}

message RecordRes {
  int32 code = 1;
  string id = 2;
}
